wal/encoder.go

```diff
@@ -39,9 +39,9 @@ type encoder struct {
 	uint64buf []byte
 }
 
-func newEncoder(w io.Writer, prevCrc uint32) *encoder {
+func newEncoder(w io.Writer, offset int, prevCrc uint32) *encoder {
```

Comment:
Passing the offset into this is evidently error-prone. Have a second function `newFileEncoder(f *os.File, prevCrc uint32) *encoder` instead of adding the offset to this function?